src/controllers/beneficial.owner.type.controller.ts

```diff
@@ -82,7 +83,12 @@ const getNextPage = (req: Request): string => {
 
     return nextPageUrl;
   } else if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.otherLegal) {
-    return config.BENEFICIAL_OWNER_OTHER_URL;
+    let nextPageUrl = config.BENEFICIAL_OWNER_OTHER_URL;
+    if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL)){
```

Comment:
We already check if the flag is active in this function. Code can be refactored to check the flag for all urls. Otherwise we'll end up checking for the flag for each url.
e.g.
```
 if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL)){
    if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.individual) {
      return getUrlWithParamsToPath(config.BENEFICIAL_OWNER_INDIVIDUAL_WITH_PARAMS_URL, req);
    } else if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.otherLegal) {
      return getUrlWithParamsToPath(config.BENEFICIAL_OWNER_OTHER_WITH_PARAMS_URL, req);
    } ...
  } else {
    if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.individual) {
      return config.BENEFICIAL_OWNER_INDIVIDUAL_URL;
    } else if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.otherLegal) {
      return config.BENEFICIAL_OWNER_OTHER_URL;
    } ...
}
```

Reply:
The trouble is we then end up checking this twice:
if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.individual)
so I'm not sure what we gain.

Reply:
By checking the flag only once we have all the feature flagged code in one place, which will make it easier when we come to remove the flag in the future. Otherwise we'll eventually have 5 flag blocks of code to remove - one for each URL in this function.

Reply:
Just realised this before I read your reply have updated this into blocks as shown.